dbt/include/spark/macros/materializations/table.sql

```diff
@@ -17,15 +16,24 @@
     {{ adapter.drop_relation(old_relation) }}
   {%- endif %}
 
-  -- build model
-  {% call statement('main') -%}
-    {{ create_table_as(False, target_relation, sql) }}
-  {%- endcall %}
-  
+  {%- call statement('main', language=language) -%}
+    {{ create_table_as(False, target_relation, compiled_code, language) }}
+  {%- endcall -%}
+
   {% do persist_docs(target_relation, model) %}
 
   {{ run_hooks(post_hooks) }}
 
   {{ return({'relations': [target_relation]})}}
 
 {% endmaterialization %}
+
+
+{% macro py_complete_script(compiled_code, target_relation) %}
```

Comment:
Is this only for `dbt-spark`(`dbt-databricks`)? I guess we should have this in `dbt-core` with dispatch pattern?